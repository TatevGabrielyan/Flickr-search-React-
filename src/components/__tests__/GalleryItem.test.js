import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import GalleryItem from '../GalleryItem';

describe('GalleryItem component', () => {
  const id = '123456789';
  const title = 'This is a title!';
  const url_q =
    'https://farm2.staticflickr.com/1943/43652580650_283f870b06_q.jpg';
  const setIdDetail = jest.fn();
  const component = (
    <GalleryItem
      id={id}
      title={title}
      url_q={url_q}
      setIdDetail={setIdDetail}
    />
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click on immage', () => {
    const tree = TestUtils.renderIntoDocument(component);
    const image = TestUtils.findRenderedDOMComponentWithTag(tree, 'img');
    TestUtils.Simulate.click(image);
    expect(setIdDetail).toHaveBeenCalled();
    expect(setIdDetail).toHaveBeenCalledTimes(1);
    expect(setIdDetail).toBeCalledWith(id);
  });
});
