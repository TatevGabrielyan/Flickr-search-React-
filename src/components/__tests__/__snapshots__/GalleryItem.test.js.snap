// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem component Snapshot matchs 1`] = `
<div
  className="GalleryItem"
>
  <img
    alt="This is a title!"
    className="GalleryImage"
    onClick={[Function]}
    src="https://farm2.staticflickr.com/1943/43652580650_283f870b06_q.jpg"
  />
</div>
`;
